python/ambassador_diag/diagd.py

```diff
@@ -1520,6 +1520,13 @@ def _load_ir(self, rqueue: queue.Queue, aconf: Config, fetcher: ResourceFetcher,
         with self.app.aconf_timer:
             aconf.load_all(fetcher.sorted())
 
+            # TODO(Flynn): This is an awful hack. Have aconf.load(fetcher) that does
+            # this correctly.
+            #
+            # I'm not doing this at this moment because aconf.load_all() is called in a
+            # lot of places, and I don't want to destablize 2.2.2.
+            aconf.load_invalid(fetcher)
```

Comment:
Actually the resource fetcher in this call does not have `invalid` initialized at the time of this call. It seems to only be initialized by `fetcher.py - parse_watt()` and not by `__init__` (and parse_watt() was evidently not called by this point.

Reply:
Which leads to the `'ResourceFetcher' object has no attribute 'invalid'` test error when `load_invalid` does `self.invalid = fetcher.invalid`

Reply:
Hmmm, that's in `_load_ir` -- `parse_watt` will be called above, in `load_config_watt`, before `_load_ir` is called. No other paths ~are~should be used any more.

OTOH, it's hardly a bad thing to initialize it in `__init__`. Done.

Reply:
(Of course I wrote that _before_ looking at the test results. 🙄 )